### PR TITLE
Add --python-disable-dependency option to disable specific python dependencies

### PR DESCRIPTION
**Pull request by cure**
This is useful for example in the case of a dependency on the python
PyYAML package.

By default, this dependency gets translated to a debian package
dependency on python-pyyaml. That package does not exist on Debian
systems (it is called python-yaml), so a way to override the dependency is needed.

A similar option exists already for gems (--gem-disable-dependency).

### CODE REVIEW

**Comment by cure** [comment]
Ping? Any love for this pull request? I'd really like to upstream this functionality so I don't have to depend on my special fpm version.

Thanks,
Ward.

File: lib/fpm/package/python.rb [hunk 1]
```diff
@@ -242,7 +249,7 @@ def load_package_info(setup_py)
         # convert dependencies from python-Foo to python-foo
         name = name.downcase if attributes[:python_downcase_dependencies?]
         "#{name} #{cmp} #{version}"
-      end
+      end.compact!
```
Comment (jordansissel) [inline lib/fpm/package/python.rb]: What does this fix?

File: lib/fpm/package/python.rb [hunk 2]
```diff
@@ -228,6 +231,10 @@ def load_package_info(setup_py)
         end
         name, cmp, version = match.captures
 
+        if attributes[:python_disable_dependencies]
+          next if attributes[:python_disable_dependencies].include?(name)
```
Comment (jordansissel) [inline lib/fpm/package/python.rb]: the outer `if` here can be removed, I think, if you set the `:default => []` in the option

**Comment by jordansissel** [comment]
I haven't tested this yet, and there are no tests for this. Can you write some tests for this?
Can someone test this before we merge it?

Otherwise, I left some comments inline.

**Comment by cure** [comment]
Thanks for your comments. I've addressed them all, and added a number of tests. Does this look better?

**Comment by cure** [comment]
ping? Sorry to keep bothering you - does the modified patch look better?

**Comment by jordansissel** [comment]
I have not yet had time or energy to work on this.

**Comment by jordansissel** [comment]
Alrighty! Code looks good. Tests pass. Thank you so much :)

I appreciate your patience and efforts! Sorry for the delays.

**Pull request merged**